### PR TITLE
add a little ajax update

### PR DESCRIPTION
**Pull request by Sbonelo01**
Related issues: https://www.notion.so/510ac24a05154e0e84522231f57ebcd5?v=871a13b6f14944f18a1f14fbd369caad&p=7f0d63e948114468851b2ac84f5b126c&pm=s

## I solemnly swear that:

- [x] I ran the hugo server and looked at my changed in the browser with my own eyes
- [x] I ran the linter and there were no errors
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code

### CODE REVIEW

File: content/projects/nodejs/ajax/_index.md [hunk 1]
```diff
@@ -45,10 +45,14 @@ Also, make sure that if you create any new visitors then they are visible on the
...
 ## Add a visitor
 
-Add a button labelled "New Visitor". This button should redirect the user to the new visitor form that was created in an earlier step in this project. 
+Add a button labelled "New Visitor". This button should redirect the user to the new visitor form that was created in an earlier step in this project.
 
 Update the "thank you page" so that it only displays for 3 seconds and then redirects the user back to the visitors table. Of course the new visiter should be visible in the table.
 
+## Up for a challenge?
+
+So far, you've made use of only 2 API end points(`listAllVisitors` and `deleteVisitor`). Can you figure out how to make use the `updateVisitor` endpoint? The implementation here is completely up to you.
```
Comment (Ngoakor12) [inline content/projects/nodejs/ajax/_index.md]: - The endpoints you are referencing here were changed to follow "RESTful API design" a while back. [See](http://syllabus.africacode.net/projects/nodejs/api/index.html)

File: content/topics/intro-to-ajax/_index.md [hunk 2]
```diff
@@ -3,8 +3,18 @@ _db_id: 146
...
-- ajax
+  - ajax
 title: Introduction to Ajax
 ---
 
-[Ajax Tutorial](https://www.codementor.io/sheena/ajax-tutorial-web-development-du107rzaq?referral=sheena-kvo1e6ewh)
\ No newline at end of file
+Ajax, short for Asynchronous JavaScript and XML, revolutionized web development by enabling seamless asynchronous communication between a web application's frontend and server. This technology allows specific parts of a webpage to be updated dynamically without requiring a complete refresh, resulting in faster and more responsive user experiences. By utilizing JavaScript to send and receive data in the background, Ajax has significantly contributed to the interactivity and real-time capabilities of modern web applications.
+
+Have a look at the following tutorial to see how ajax functionality can be achieved with JQuery:
```
Comment (Ngoakor12) [inline content/topics/intro-to-ajax/_index.md:12]: Maybe this could be a heading?

File: content/topics/intro-to-ajax/_index.md [hunk 3]
```diff
@@ -3,8 +3,18 @@ _db_id: 146
...
+
+[Ajax Tutorial with JQuery](https://www.codementor.io/sheena/ajax-tutorial-web-development-du107rzaq?referral=sheena-kvo1e6ewh)
+
+Ajax functionality can also be achieved with vanilla JavaScript. This involves using the XMLHttpRequest object to send asynchronous HTTP requests to the server and handle responses without requiring additional libraries. Through this approach, developers can update webpage content dynamically and enhance user interactions without relying on external frameworks.
+
+Have a look at the following tutorial to see how ajax functionality can be achieved with vanilla JavaScript:
```
Comment (Ngoakor12) [inline content/topics/intro-to-ajax/_index.md:18]: Also this one a heading too?

File: content/topics/intro-to-ajax/_index.md [hunk 4]
```diff
@@ -3,8 +3,18 @@ _db_id: 146
...
+
+[Ajax Tutorial with vanilla JS](https://developer.mozilla.org/en-US/docs/Web/Guide/AJAX)
```
Comment (Ngoakor12) [inline content/topics/intro-to-ajax/_index.md:20]: Another way of making "ajax" calls is with `fetch` web api or a library like `axios`, maybe that can be added under the "other ways" section. The section can be named whatever you feel is a good match

**Review by Ngoakor12 — CHANGES_REQUESTED** [review]
- Updating this heading `Delete visitors and update the table` to `Delete visitors` removes the confusion being experienced by learners

File: content/projects/nodejs/ajax/_index.md [hunk 5]
```diff
@@ -2,28 +2,28 @@
...
 tags:
-- node
-- ajax
+  - node
+  - ajax
 title: Add a little Ajax
 ---
 
-In this project you'll be creating a more modern frontend for your application. You'll need to create a whole new page that allows a user to see a list of visitors and to delete visitors.
+In this project you'll be creating a more modern frontend for your application. You'll need to create a whole new page that allows a user to see a list of visitors and to delete visitors. The technology to use is entirely up to you.
```
Comment (sheenarbw) [inline content/projects/nodejs/ajax/_index.md]: This was put in so that the learners would need to think for themselves.

**Review by sheenarbw — CHANGES_REQUESTED** [review]
If the goal of this is to be specific about what technology the learners are meant to use, then it might be good to be specific about it.

That said, developers should be able to read up on things and think for themselves. 

Why not say something like:

```
## What technology to use for sending AJAX requests

As a developer, it's important that you can think for yourself and read up on things. Tech changes all the time.

Do some googling - read up on the current best practices. Find out how the pros are making requests from their client-side applications. Then make a choice. 

```

If a learner demonstrates that they lack confidence or skill then the thing to do is push them to develop that confidence or skill. Just telling them what to use all the time wont let them develop themselves.

Maybe I'm misunderstanding what you are trying to achieve here though.

File: content/topics/intro-to-ajax/_index.md (hunk 4 above)
Comment (Sbonelo01) [reply content/topics/intro-to-ajax/_index.md:20]: If one reads this Mozilla write-up, they will come across information on the fetch web API on the "Related technologies headings"

File: content/projects/nodejs/ajax/_index.md [hunk 6]
```diff
@@ -51,7 +56,7 @@ Update the "thank you page" so that it only displays for 3 seconds and then redi
 
 ## Up for a challenge?
 
-So far, you've made use of only 2 API end points(`listAllVisitors` and `deleteVisitor`). Can you figure out how to make use the `updateVisitor` endpoint? The implementation here is completely up to you.
+So far, you've made use of only 3 API end points(`POST`, `GET` and `DELETE`). Can you figure out how to make use the `PUT` endpoint? The implementation here is completely up to you.
```
Comment (TheMaja) [inline content/projects/nodejs/ajax/_index.md]: for grammar i will rephrase to"So far, you've utilized only three API endpoints (POST, GET, and DELETE). Can you figure out how to use the PUT endpoint? The implementation here is entirely up to you"

**Review by TheMaja — CHANGES_REQUESTED** [review]
- also change the `visiter` on line 55. otherwise, it looks good to me.

**Review by Ngoakor12 — APPROVED** [review]

File: content/projects/nodejs/ajax/_index.md [hunk 7]
```diff
@@ -45,10 +50,14 @@ Also, make sure that if you create any new visitors then they are visible on the
...
 
-Add a button labelled "New Visitor". This button should redirect the user to the new visitor form that was created in an earlier step in this project. 
+Add a button labelled "New Visitor". This button should redirect the user to the new visitor form that was created in an earlier step in this project.
+
+Update the "thank you page" so that it only displays for 3 seconds and then redirects the user back to the visitors table. Of course the new visitor should be visible in the table.
+
+## Up for a challenge?
 
-Update the "thank you page" so that it only displays for 3 seconds and then redirects the user back to the visitors table. Of course the new visiter should be visible in the table.
+So far, you've utilized only three API end points(`POST`, `GET` and `DELETE`). Can you figure out how to make use the `PUT` endpoint? The implementation here is completely up to you.
```
Comment (sheenarbw) [inline content/projects/nodejs/ajax/_index.md]: This language is a bit odd. Those aren't endpoints. Those are verbs.

How about just saying something like:

"""
We can now add, list and delete visitor objects.  Add a feature that lets the user update a Visitor's details. Pay close attention to usability - how can you make this feature easy and obvious to the user?
"""

**Review by sheenarbw — CHANGES_REQUESTED** [review]
There's one thing that can be reworded.

**Review by sheenarbw — APPROVED** [review]

**Pull request merged**